### DISCUSSION TITLE
How to get Id of an element using the text

### DISCUSSION THREAD

**Discussion by kamalsivalingam** [mobile-dev-inc/Maestro#1757] [Q&A]
Hi,
Is there a way to get the Id of an element using the text in maestro flow. We are calling another javascript function that needs id. However i couldnt find a way to get an elements id using the text. This element is dynamically added and the Id is the new guid

**Comment by portyu9** [comment]
Hi @kamalsivalingam 

There currently isn't a Maestro flow command that means:

> find an element by text, then expose that matched element's `id` to JavaScript.

`copyTextFrom` only copies the element's **text** into `maestro.copiedText`; it doesn't expose selector metadata such as the accessibility/resource ID.

If the generated GUID is part of the element's ID and you only need it for selecting the element, remember that Maestro IDs support regex matching, so you may be able to select it without extracting the GUID.

If your JavaScript genuinely needs that GUID as data, the more reliable design is to expose/pass the value separately — for example through test data, an API response, environment variable, or accessible text — rather than scraping Maestro's internal accessibility hierarchy at runtime.

`maestro hierarchy` is useful for debugging the actual hierarchy externally, but it does not turn an element's ID into a flow variable.

Please mark this answer as accepted if it helped, thank you.
